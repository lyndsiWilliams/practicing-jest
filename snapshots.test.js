var itemStock = [
  { 'id': '1', 'ItemName': 'Razors', 'Stock': '10' },
  { 'id': '2', 'ItemName': 'Socks', 'Stock': '1' },
  { 'id': '3', 'ItemName': 'Towels', 'Stock': '100' },
  { 'id': '4', 'ItemName': 'Socks', 'Stock': '1000' },
];

function filterItemStock(arr, key, term) {
  return arr.filter(function(obj) {
    return obj[key] === term;
  });
};

test('Returns all items with matching id', () => {
  expect(filterItemStock(itemStock, 'id', '1')).toEqual([
    { 'id': '1', 'ItemName': 'Razors', 'Stock': '10' }
  ]);
});

test('Returns all items with matching item name', () => {
  expect(filterItemStock(itemStock, 'ItemName', 'Socks')).toEqual([
    { 'id': '2', 'ItemName': 'Socks', 'Stock': '1' },
    { 'id': '4', 'ItemName': 'Socks', 'Stock': '1000' },  
  ]);
});


test('Returns all items with matching id', () => {
  expect(filterItemStock(itemStock, 'id', '1')).toMatchSnapshot();
});

test('Returns all items with matching item name', () => {
  expect(filterItemStock(itemStock, 'ItemName', 'Socks')).toMatchSnapshot();
});