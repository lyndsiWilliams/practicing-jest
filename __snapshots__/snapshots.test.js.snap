// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Returns all items with matching id 1`] = `
Array [
  Object {
    "ItemName": "Razors",
    "Stock": "10",
    "id": "1",
  },
]
`;

exports[`Returns all items with matching item name 1`] = `
Array [
  Object {
    "ItemName": "Socks",
    "Stock": "1",
    "id": "2",
  },
  Object {
    "ItemName": "Socks",
    "Stock": "1000",
    "id": "4",
  },
]
`;
